.dotnet/Azure.AI.OpenAI.Staging/Custom/AzureOpenAIClient.cs

```diff
@@ -0,0 +1,144 @@
+            Pipeline,
+            _cachedOptions?.Endpoint ?? GetConfiguredEndpoint(),
+            _cachedOptions as AzureOpenAIClientOptions);
+    }
+
+    public override EmbeddingClient GetEmbeddingClient(string deploymentName)
+    {
+        return new AzureEmbeddingClient(
+            deploymentName,
+            Pipeline,
+            _cachedOptions?.Endpoint ?? GetConfiguredEndpoint(),
+            _cachedOptions as AzureOpenAIClientOptions);
+    }
+
+    public override FileClient GetFileClient()
+    {
+        throw new NotImplementedException();
+    }
+
+    public override FineTuningClient GetFineTuningClient()
+    {
+        return new AzureFineTuningClient(
+            Pipeline,
+            _cachedOptions?.Endpoint ?? GetConfiguredEndpoint(),
+            _cachedOptions as AzureOpenAIClientOptions);
+    }
+
+    public override ImageClient GetImageClient(string deploymentName)
+    {
+        return new AzureImageClient(
+            deploymentName,
+            Pipeline,
+            _cachedOptions?.Endpoint ?? GetConfiguredEndpoint(),
+            _cachedOptions as AzureOpenAIClientOptions);
+    }
+
+    public override ModelManagementClient GetModelManagementClient()
+    {
+        throw new NotSupportedException($"Azure OpenAI does not support the OpenAI model management API. "
+            + "Please use the Azure AI Services Account Management API to interact with Azure OpenAI model deployments.");
+    }
+
+    public override ModerationClient GetModerationClient()
+    {
+        throw new NotSupportedException($"Azure OpenAI does not support the OpenAI moderations API. "
+            + "Please refer to the documentation on Microsoft's Responsible AI embedded content filters to learn "
+            + "more about Azure OpenAI's content filter policies and response annotations.");
+    }
+
+    private static ClientPipeline CreatePipeline(PipelinePolicy authenticationPolicy, OpenAIClientOptions options = null)
+        => ClientPipeline.Create(
+            options ?? new(),
+            perCallPolicies: [],
+            perTryPolicies: [authenticationPolicy],
+            beforeTransportPolicies: []);
+
+    internal static ClientPipeline CreatePipeline(ApiKeyCredential credential, OpenAIClientOptions options = null)
+    {
+        credential ??= GetConfiguredKeyCredential();
+        return CreatePipeline(ApiKeyAuthenticationPolicy.CreateHeaderApiKeyPolicy(credential, "api-key"), options);
+    }
+
+    internal static ClientPipeline CreatePipeline(TokenCredential credential, OpenAIClientOptions options = null)
+    {
+        // To do: proper token conversion from Azure.Core to System.ClientModel
+        throw new NotImplementedException();
+    }
+
+    internal static ApiKeyCredential GetConfiguredKeyCredential()
+    {
+        return new(Environment.GetEnvironmentVariable("AZURE_OPENAI_API_KEY"));
+    }
+    internal static Uri GetConfiguredEndpoint(AzureOpenAIClientOptions options = null)
```

Comment:
I think we really need to think through how this will integrate with IConfiguration when we add it in the future.

Reply:
I agree. I'll add a similar note to this use of the environment variable as I'd placed on the other in the interim:
```csharp
        // To do: IConfiguration
        configuredEndpoint ??= Environment.GetEnvironmentVariable("AZURE_OPENAI_ENDPOINT");
        if (configuredEndpoint is null)
        {
            throw new InvalidOperationException(
                $"An Azure OpenAI resource endpoint URI must be provided via the client constructor, application "
                + "configuration, or AZURE_OPENAI_ENDPOINT environment variable.");
        }
```